controllers/models/Products.js

```diff
@@ -195,35 +199,17 @@ var Products = {
 
     // returns all available products
     GetExistingProducts : function() {
-        return cleanUpProducts(filterForExistingProducts(this.products));
+        return filterForExistingProducts(this.products);
     },
 
     // returns all products, also the once that have been deleted and are no longer sold
     GetAllProducts : function() {
-        return cleanUpProducts(this.products);
+        return this.products;
```

Comment:
Why is the step of "creates a copy of the products-list that contains only the keys listed as public above" no longer necessary?

Reply:
The filtering is unnecessary because the filter contains all attributes of a product.
When I call GetAllProducts, I expect to get the products and not a copy of the actual products. (same for GetExistingProducts)